x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/form.tsx

```diff
@@ -156,3 +160,84 @@ export const useRuleIndexPattern = ({
   }, [dataSourceType, isIndexPatternLoading, data, dataViewId, initIndexPattern]);
   return { indexPattern, isIndexPatternLoading, browserFields };
 };
+
+export interface UseRuleFormsErrors {
+  defineStepForm?: FormHook<DefineStepRule, DefineStepRule>;
+  aboutStepForm?: FormHook<AboutStepRule, AboutStepRule>;
+  scheduleStepForm?: FormHook<ScheduleStepRule, ScheduleStepRule>;
+  actionsStepForm?: FormHook<ActionsStepRule, ActionsStepRule>;
+}
+
+const getFieldErrorMessages = (fieldError: ValidationError) => {
+  if (fieldError.message.length > 0) {
+    return [fieldError.message];
+  } else if (Array.isArray(fieldError.messages)) {
+    return fieldError.messages.map((message) => JSON.stringify(message));
```

Comment:
why message is stringified here but not at 173 line?

Maybe simple unit tests or comment can be helpful

Reply:
Added comment